src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2724,6 +2724,31 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
         {
             return null;
         }
+
+        public override BoundNode VisitSwitchExpression(BoundSwitchExpression node)
+        {
+            VisitRvalue(node.GoverningExpression);
+            var dispatchState = this.State;
+            var endState = UnreachableState();
+            foreach (var section in node.SwitchSections)
+            {
+                SetState(dispatchState.Clone());
+                VisitPattern(node.GoverningExpression, section.Pattern);
+                SetState(StateWhenTrue);
+                if (section.Guard != null)
+                {
+                    VisitCondition(section.Guard);
+                    SetState(StateWhenTrue);
+                }
+
+                VisitRvalue(section.Value);
```

Comment:
I can't remember the details of how this function works, so maybe it's just a misnomer, but is `section.Value required to be an r-value? Is there anything preventing us from allowing switch l-value expression? Could be useful if we decide to add a ref-switch.

Reply:
At the moment it is required to be an rvalue. I'm open to extending it, but it would require some kind of syntax change.

---
In reply to: [166482551](https://github.com/dotnet/roslyn/pull/23357#discussion_r166482551) [](ancestors = 166482551)